zaas-service/src/test/java/org/zowe/apiml/zaas/security/service/token/OIDCTokenProviderTest.java

```diff
@@ -0,0 +1,278 @@
+import org.springframework.test.util.ReflectionTestUtils;
+import org.zowe.apiml.constants.ApimlConstants;
+import org.zowe.apiml.zaas.cache.CachingServiceClientException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class OIDCTokenProviderTest {
+
+    private static final String OKTA_JWKS_RESOURCE = "test_samples/okta_jwks.json";
+
+    private static final String EXPIRED_TOKEN = "eyJraWQiOiJMY3hja2tvcjk0cWtydW54SFA3VGtpYjU0N3J6bWtYdnNZVi1uYzZVLU40IiwiYWxnIjoiUlMyNTYifQ.eyJ2ZXIiOjEsImp0aSI6IkFULlExakp2UkZ0dUhFUFpGTXNmM3A0enQ5aHBRRHZrSU1CQ3RneU9IcTdlaEkiLCJpc3MiOiJodHRwczovL2Rldi05NTcyNzY4Ni5va3RhLmNvbS9vYXV0aDIvZGVmYXVsdCIsImF1ZCI6ImFwaTovL2RlZmF1bHQiLCJpYXQiOjE2OTcwNjA3NzMsImV4cCI6MTY5NzA2NDM3MywiY2lkIjoiMG9hNmE0OG1uaVhBcUVNcng1ZDciLCJ1aWQiOiIwMHU5OTExOGgxNmtQT1dBbTVkNyIsInNjcCI6WyJvcGVuaWQiXSwiYXV0aF90aW1lIjoxNjk3MDYwMDY0LCJzdWIiOiJzajg5NTA5MkBicm9hZGNvbS5uZXQiLCJncm91cHMiOlsiRXZlcnlvbmUiXX0.Cuf1JVq_NnfBxaCwiLsR5O6DBmVV1fj9utAfKWIF1hlek2hCJsDLQM4ii_ucQ0MM1V3nVE1ZatPB-W7ImWPlGz7NeNBv7jEV9DkX70hchCjPHyYpaUhAieTG75obdufiFpI55bz3qH5cPRvsKv0OKKI9T8D7GjEWsOhv6CevJJZZvgCFLGFfnacKLOY5fEBN82bdmCulNfPVrXF23rOregFjOBJ1cKWfjmB0UGWgI8VBGGemMNm3ACX3OYpTOek2PBfoCIZWOSGnLZumFTYA0F_3DsWYhIJNoFv16_EBBJcp_C0BYE_fiuXzeB0fieNUXASsKp591XJMflDQS_Zt1g";
+
+    private static String VALID_TOKEN;
+    private static final String MALFORMED_TOKEN = "token";
+    private static JWKSet localJwkSet;
+    private static String oktaJwks;
+
+    private OIDCTokenProvider oidcTokenProvider;
+
+    @Mock
+    private DefaultResourceRetriever resourceRetriever;
+    @Mock
+    private CloseableHttpClient httpClient;
+
+    static Stream<String> invalidTokens() {
+        return Stream.of(
+            EXPIRED_TOKEN, MALFORMED_TOKEN, "", null
+        );
+    }
+
+    @BeforeAll
+    static void init() throws Exception {
+        var now = Instant.now();
+        var pKey = loadPrivateKey("../keystore/localhost/localhost.keystore.p12", "localhost", "password");
+        VALID_TOKEN = Jwts.builder()
+            .header().keyId("0987").and()
+            .subject("user")
+            .issuedAt(Date.from(now))
+            .expiration(Date.from(now.plusSeconds(1200)))
+            .issuer("API ML")
+            .id(UUID.randomUUID().toString())
+            .signWith(pKey, Jwts.SIG.RS256).compact();
+    }
+
+    static PrivateKey loadPrivateKey(String path, String alias, String password) throws Exception {
```

Comment:
parameter can be inline:

```java
 static PrivateKey loadPrivateKey() throws Exception {
        KeyStore ks = KeyStore.getInstance("PKCS12");
        try (FileInputStream fis = new FileInputStream("../keystore/localhost/localhost.keystore.p12")) {
            ks.load(fis, "password".toCharArray());
        }
        Key key = ks.getKey("localhost", "password".toCharArray());
        var cert = ks.getCertificate("localhost");
        var pubKey = cert.getPublicKey();
        if (pubKey instanceof RSAPublicKey rsaPublicKey) {
            var k = new RSAKey.Builder(rsaPublicKey).keyID("0987").build().toPublicJWK();
            localJwkSet = new JWKSet(k);
        }

        return (PrivateKey) key;
    }
```

Reply:
The current implementation allows for future reuse